src/Bicep.Cli/Commands/BuildParamsCommand.cs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Bicep.Cli.Arguments;
+using Bicep.Cli.Logging;
+using Bicep.Cli.Services;
+using Bicep.Core.Configuration;
+using Bicep.Core.Emit;
+using Bicep.Core.Features;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Semantics;
+using Microsoft.Extensions.Logging;
+
+namespace Bicep.Cli.Commands
+{
+    public class BuildParamsCommand : ICommand
+    {
+        private readonly ILogger logger;
+        private readonly IDiagnosticLogger diagnosticLogger;
+        private readonly IOContext io;
+        private readonly CompilationService compilationService;
+        private readonly CompilationWriter writer;
+        private readonly IFeatureProviderFactory featureProviderFactory;
+
+        public BuildParamsCommand(
+            ILogger logger,
+            IDiagnosticLogger diagnosticLogger,
+            IOContext io,
+            CompilationService compilationService,
+            CompilationWriter writer,
+            IFeatureProviderFactory featureProviderFactory)
+        {
+            this.logger = logger;
+            this.diagnosticLogger = diagnosticLogger;
+            this.io = io;
+            this.compilationService = compilationService;
+            this.writer = writer;
+            this.featureProviderFactory = featureProviderFactory;
+        }
+
+        public async Task<int> RunAsync(BuildParamsArguments args)
+        {
+            var paramsInputPath = PathHelper.ResolvePath(args.ParamsFile);
+            var bicepInputPath = args.BicepFile != null ? PathHelper.ResolvePath(args.BicepFile) : null;
+
+            var features = featureProviderFactory.GetFeatureProvider(PathHelper.FilePathToFileUrl(paramsInputPath));
+            var emitterSettings = new EmitterSettings(features);
+
+            if (emitterSettings.EnableSymbolicNames)
+            {
+                logger.LogWarning(CliResources.SymbolicNamesDisclaimerMessage);
+            }
+
+            if (features.ResourceTypedParamsAndOutputsEnabled)
+            {
+                logger.LogWarning(CliResources.ResourceTypesDisclaimerMessage);
+            }
+
+            if(bicepInputPath != null && !IsBicepFile(bicepInputPath))
+            {
+                throw  new InvalidOperationException($"{bicepInputPath} is not is bicep file");
+            }
+
+
+            if(features.ParamsFilesEnabled && IsBicepparamsFile(paramsInputPath)) 
```

Comment:
Since this is a new command, you can simplify some of this logic. Running `bicep build-params` without the params files feature enabled shouldn't be allowed, and the command should fail if the file provided isn't a bicep params file. E.g.,

```csharp
if (!features.ParamsFilesEnabled)
{
    logger.LogError("The X feature must be enabled in order to use the 'build-params' command" or similar);
    return 1;
}

if (!IsBicepparamsFile(paramsInputFile))
{
    logger.LogError("X is not a bicep params file" or similar);
    return 1;
}

// proceed with compilation
```